### PR TITLE
Fixes to ci

### PR DESCRIPTION
**Pull request by sbcgua**
Mainly addressing this: https://github.blog/changelog/2022-10-11-github-actions-deprecating-save-state-and-set-output-commands/
And some minor improvements (better logging) to the push-tags script

But also I'd like to report on the `auto-tag-artifact` job issues progress. So it still does not work. And it is very difficult to debug without an almost complete replication of the repos structure. So ! It seems that the issue was with the DEPLOY_TOKEN, which (i assume) contained some special symbols. Though this is strange because the build deployment works with the same command ... but the existing command works with a simple "XXXX" as a token ... so really unclear. And I don't want to print token in the console to debug it for the security reasons ...
Anyway, I made some improvements that,  hopefully will work next time for the next tag

### CODE REVIEW

File: .github/workflows/main-build.yml [hunk 1]
```diff
@@ -73,6 +73,7 @@ jobs:
         GIT_USER_NAME: 'github-actions[bot]'
         GIT_USER_EMAIL: 'github-actions[bot]@users.noreply.github.com'
       run: ./ci/push-tag.sh ${{ needs.auto-tag.outputs.pushedTag }}
+      # Importantly, this calls the push-tags script from build repositayr, not the local one !
```
Comment (mbtools) [inline .github/workflows/main-build.yml:76]: repository

**Comment by mbtools** [comment]
As you wrote in your comment it will run `push-tag.sh` from the build repo (ie 
`https://github.com/abapGit/build/blob/main/ci/push-tag.sh`). So why do we have another `push-tag.sh` in this repo (ie `https://github.com/abapGit/abapGit/blob/main/ci/push-tag.sh`)?

File: .github/workflows/main-build.yml (hunk 1 above)
Comment (sbcgua) [reply .github/workflows/main-build.yml]: fixed

**Comment by sbcgua** [comment]
Yeah, hmm, fair question ... so one subtle bug was that to tag the `build` the script clones `build` repo. Which doesn't have CI scripts in it. So when the script failed I added the `push-tag.sh` directly to `build` and had an opportunity to re-run the failed script 2-3 times. Yet at some point the failed script (`auto-tag-artifact`) succeeded and I cannot rerun only it alone again. Thus I still don't know for sure if it works :( Debugging this thing is very tricky (or requires too much free time, I just don't have at the moment).

But yes ! This is a feature-like bug. I think the script should really clone BOTH repositories to have access to the script from abapGit repo. I didn't do it because I wanted to have some flexibility for the next tag. If it also fails ... I will have an attempt or 2 to adjust the script in `build repo`. Not to spam the AG source repo with PRs. And to be able to run for the "commit with a new version". And then, switch to running script from AG repository. If you insist I can try switching it right now, no prob. Just that the next try might also be unsuccessful.

Sorry if it looks complicated. It just is. When I starting some devops stuff for my repos, there are just 5-7-10 commits directly to master in a row to finally ignite it ... and in AG I cannot do this.

**Comment by mbtools** [comment]
I had a similar experience trying to get something on GH to work. After too many test commits, I ended up trying with a new set of repos.

The current CI setup is still fairly limited but we might want to consider a clone for test purposes. We have abapGit-test org and could add abapGit-test/abapGit and abapGit-test/build to it.

**Comment by sbcgua** [comment]
yeap, exactly.

abapGit-test/abapGit: yeah, kind of QA environment. And copy it from prod once per half a year :))))))

**Comment by sbcgua** [comment]
OK now push tag starts from ag repo. Let's try ...eventually

**Comment by sbcgua** [comment]
unit tests check is irrelevant to this PR ...

**Comment by larshp** [comment]
update branch to fix unit test

**Pull request merged**